imports/client/components/PuzzleListPage.tsx

```diff
@@ -424,37 +428,52 @@ interface PuzzleListPageProps extends PuzzleListPageWithRouterParams {
   canUpdate: boolean;
   allPuzzles: PuzzleType[];
   allTags: TagType[];
+  hunt?: HuntType;
 }
 
 class PuzzleListPage extends React.Component<PuzzleListPageProps> {
   render() {
+    const leadLinks = this.props.hunt?.homepageUrl && (
+      <p className="lead">
+        <a href={this.props.hunt.homepageUrl} target="_blank" rel="noopener noreferrer" title="Open the hunt homepage">
+          <FontAwesomeIcon icon={faExternalLinkAlt} />
+          {' '}
+          Hunt homepage
+        </a>
+      </p>
+    );
     if (!this.props.ready) {
       return <span>loading...</span>;
     } else {
       return (
-        <PuzzleListView
-          match={this.props.match}
-          history={this.props.history}
-          location={this.props.location}
-          huntId={this.props.match.params.huntId}
-          canAdd={this.props.canAdd}
-          canUpdate={this.props.canUpdate}
-          puzzles={this.props.allPuzzles}
-          allTags={this.props.allTags}
-        />
+        <div>
+          {leadLinks}
+          <PuzzleListView
+            match={this.props.match}
+            history={this.props.history}
+            location={this.props.location}
+            huntId={this.props.match.params.huntId}
+            canAdd={this.props.canAdd}
+            canUpdate={this.props.canUpdate}
+            puzzles={this.props.allPuzzles}
+            allTags={this.props.allTags}
+          />
+        </div>
       );
     }
   }
 }
 
 const PuzzleListPageContainer = withTracker(({ match }: PuzzleListPageWithRouterParams) => {
+  const huntHandle = Meteor.subscribe('mongo.hunts', { _id: match.params.huntId });
```

Comment:
No need to subscribe here -- we've already subscribed to `mongo.hunts` with that `huntId` up in HuntApp.tsx, which is higher in the tree.  By the time we're rendering PuzzleListPage, we're guaranteed we've already got the hunt.

So remove this line, don't worry about tracking if it's ready or not, and just always pass the hunt down.

Reply:
I was wondering how we felt about that redundancy in the name of modularity. It's not hard to imagine a change breaking this assumption, but I think I'm ok with it.